sdk/synapse/synapse-artifacts/review/synapse-artifacts.api.md

```diff
@@ -126,12 +132,27 @@ export type AmazonRedshiftTableDataset = Dataset & {
     schemaTypePropertiesSchema?: any;
 };
 
+// @public
+export type AmazonS3Dataset = Dataset & {
+    type: "AmazonS3Object";
+    bucketName: any;
+    key?: any;
+    prefix?: any;
+    version?: any;
+    modifiedDatetimeStart?: any;
+    modifiedDatetimeEnd?: any;
```

Comment:
Are these really `any` types, or are these the `oneOf` types you mentioned in your post? Some things like bucketName, key, prefix, and even version I wouldn't expect to be anything other than a string.

Reply:
Talked with @joheredi offline. These are `oneOf` types that will eventually be modeled as `string | Foo` (expression?)

I'd still like to understand what the expression would look like for these types but given this is a preview and autorest is making improvements on how to model these, I don'tt think this is blocking.

Reply:
Thanks @chradek! For context here is some information on what this expression is:
https://github.com/Azure/azure-rest-api-specs/pull/12924#issuecomment-797200276